docs/labels.md

```diff
@@ -11,50 +10,93 @@ key/value labels set on it, with at most one label with a particular key.
 dimensions, such as:
-   - `release=stable`, `release=canary`, ...
-   - `environment=dev`, `environment=qa`, `environment=production`
-   - `tier=frontend`, `tier=backend`, ...
-   - `partition=customerA`, `partition=customerB`, ...
-   - `track=daily`, `track=weekly`
+   - `"release" : "stable"`, `"release" : "canary"`, ...
+   - `"environment" : "dev"`, `"environment" : "qa"`, `"environment" : "production"`
+   - `"tier" : "frontend"`, `"tier" : "backend"`, `"tier" : "middleware"`
+   - `"partition" : "customerA"`, `"partition" : "customerB"`, ...
+   - `"track" : "daily"`, `"track" : "weekly"`
 
 These are just examples; you are free to develop your own conventions.
 
-Label selectors permit very simple filtering by label keys and values. The API currently only supports equality label selectors that are ANDed together. For example, if you specify the selector below in a JSON config, you'll only select resources that have both the labels `environment=dev` and `tier=frontend`:
+
+## Syntax and character set
+
+As already mentioned well formed _labels_ are key value pairs. Valid label keys are comprised of two segments - prefix and name - separated by a slash (`/`).  The name segment is required and must be a DNS label: 63 characters or less, all lowercase, beginning and ending with an alphanumeric character (`[a-z0-9A-Z]`), with dashes (`-`) and alphanumerics between.  The prefix and slash are optional.  If specified, the prefix must be a DNS subdomain (a series of DNS labels separated by dots (`.`), not longer than 253 characters in total.
+If the prefix is omitted, the label key is presumed to be private to the user. System components which use labels must specify a prefix.  The `kubernetes.io` prefix is reserved for kubernetes core components.
+
+Valid label values must be shorter than 64 characters, accepted characters are (`[-A-Za-z0-9_.]`) but the first character must be  (`[A-Za-z0-9]`).
+
+## Label selectors
+
+Unlike [names and UIDs](identifiers.md), labels do not provide uniqueness. In general, we expect many objects to carry the same label(s).
+
+Sets identified by labels could be overlapping (think Venn diagrams). For instance, a service might target all pods with `"tier": "frontend"` and  `"environment" : "prod"`.  Now say you have 10 replicated pods that make up this tier.  But you want to be able to 'canary' a new version of this component.  You could set up a `replicationController` (with `replicas` set to 9) for the bulk of the replicas with labels `"tier" : "frontend"` and `"environment" : "prod"` and `"track" : "stable"` and another `replicationController` (with `replicas` set to 1) for the canary with labels `"tier" : "frontend"` and  `"environment" : "prod"` and `"track" : canary`.  Now the service is covering both the canary and non-canary pods.  But you can mess with the `replicationControllers` separately to test things out, monitor the results, etc.
```

Comment:
I'd move these 3 paragraphs to just above "Future developments", below the discussion of the use of label selectors in services and replication controllers.